plugins/statistics.koplugin/main.lua

```diff
@@ -2285,21 +2285,30 @@ function ReaderStatistics:getFirstTimestamp()
 end
 
 function ReaderStatistics:getReadingRatioPerHourByDay(month)
+    -- We used to have in the SQL statement (with ? = 'YYYY-MM'):
+    --   WHERE  strftime('%Y-%m', start_time, 'unixepoch', 'localtime') = ?
+    -- but strftime()ing all start_time is slow.
+    -- Comverting the month into timestamp boundaries, and just comparing
+    -- integers, can be 5 times faster.
+    -- We let SQLite compute these timestamp boundaries from the provided
+    -- month; we need the start of the month to be a real date:
+    month = month.."-01"
     local sql_stmt = [[
         SELECT
             strftime('%Y-%m-%d', start_time, 'unixepoch', 'localtime') day,
             strftime('%H', start_time, 'unixepoch', 'localtime') hour,
             sum(duration)/3600.0 ratio
         FROM   page_stat
-        WHERE  strftime('%Y-%m', start_time, 'unixepoch', 'localtime') = ?
+        WHERE  start_time >= strftime('%s', ?, 'utc')
```

Comment:
a `WHERE start_time BETWEEN x AND y` would possibly be a smidge faster ;).

Reply:
Done. Not tested if it changes speed.